addon/-private/object.ts

```diff
@@ -3,16 +3,88 @@ import { assert } from '@ember/debug';
 import { USE_PROXY } from './config';
 import { consume, dirty, SUPPORTS_PROXY } from './util';
 
+const setterPropertyPrefix = '__PROPERTY__';
+
+function defineHiddenProperty(obj: object, propertyName: string, value: any) {
+  Object.defineProperty(obj, propertyName, {
+    enumerable: false,
+    configurable: true,
+    writable: true,
+    value
+  })
+}
+
+function installSetter(obj: object, keyName: string, desc: PropertyDescriptor) {
```

Comment:
I was actually thinking originally that we would follow `Object.assign` rules, but I'm open to being able to use getters and setters, let's try it out. We can remove it in a future release if it's problematic, since this is pre-v1.0. For reference, how does this work in Vue/MobX?

I'm actually not sure we really need to wrap getters and setters though, consider this:

```js
let foo = new TrackedObject({
  get bar() {
    return this.baz;
  },

  set bar(value) {
    this.baz = value;
  },

  baz: 123
});
```

This will autotrack when accessing `bar` since it is accessing `baz`, which is tracked, and will update when we update it correctly since it's invalidating `baz`. This would follow the same model as tracked getters/setters in general, so we could just forward them without wrapping.

Reply:
Vue`s - https://github.com/vuejs/vue/blob/0b31647c41b0c916d40b962723b1de06672d0e01/src/core/observer/index.js#L135

Reply:
getters setters behavour can be complex:

```js
{
  ctx: {
	baz: 123
  },
  get bar() {
    return this.ctx.baz; 
  },

  set bar(value) {
    this.ctx.baz = value;
  }
}
```

Reply:
They can be, but this somewhat violates the principles behind autotracking. We can add the notifications later on, but for now I'd rather not wrap getters/setters.

Reply:
@pzuraq Is my changes correct?
I'm not confident about understanding `I was actually thinking originally that we would follow Object.assign rules...`

Reply:
The latest changes look good, with one small tweak. It's not that we shouldn't have getters/setters, it's that they shouldn't track in general, since they represent a derived value.

Also, re: `Object.assign` rules:

```js
let foo = {
  bar: 123,
  get baz() {
    return this.bar;
  }
}

Object.assign({}, foo); // { bar: 123, baz: 123 }
```

`Object.assign` gets the value and assigns it to the new object, so the getters don't get transferred. I'd actually like the getters though, per my suggestion.